private/bufpkg/bufcheck/buflint/internal/buflintvalidate/buflintvalidate.go

```diff
@@ -0,0 +1,27 @@
+// Copyright 2020-2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package buflintvalidate
+
+import "github.com/bufbuild/buf/private/pkg/protosource"
+
+func NewValidateModule(
```

Comment:
This type isn't needed, and IMO actually detracts from clarity. We're returning a private type, meaning the Godoc attached to the single (public) method won't be displayed, and we're not actually saving anything here - you could just have a `ValidateField` public interface (with documentation), with a `NewValidateField`. `*validateModule` is just storing two parameters, and doesn't store any state - this doesn't change usability in the `check` function. Going even further, I think you could just have a function `CheckFieldRules` that takes everything, as `*validateField` isn't that necessary either, but this is more a nit.

Reply:
resolved in [182d2e2](https://github.com/bufbuild/buf/pull/2377/commits/182d2e21541d6084e75fd345150fa990cde60d62). This now raises a problem with 2 style rules:
1. `buflintvalidate.go` must exist for the package but has no natural contents
2. `validate_field.go` must contain struct definition, constructors and methods. 

This is likely a smell because it doesn't conform to the style guide correctly. Let's consider actioning your nit on `validateField` this would expand the footprint of each function signature significantly - do you hate that more or less than the validateField methods?